gobblin-tunnel/src/test/java/gobblin/tunnel/TunnelTest.java

```diff
@@ -0,0 +1,238 @@
+package gobblin.tunnel;
+
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.InetSocketAddress;
+import java.net.SocketException;
+import java.net.URL;
+import java.nio.ByteBuffer;
+import java.nio.channels.SocketChannel;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import org.apache.commons.io.IOUtils;
+import org.mockserver.integration.ClientAndServer;
+import org.mockserver.model.HttpForward;
+import org.mockserver.model.HttpRequest;
+import org.mockserver.model.HttpResponse;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertNotNull;
+import static org.testng.Assert.assertTrue;
+
+
+/**
+ * Tests tunneling through an http proxy.
+ *
+ * In order to execute these tests an http proxy must be running on localhost
+ * and listening on port 10928
+ */
+public class TunnelTest {
+
+  private ClientAndServer _mockServer;
+  public static final int PORT = 10928;
+
+  @BeforeClass
+  public void startProxy()
+      throws IOException {
+    _mockServer = ClientAndServer.startClientAndServer(PORT);
+  }
+
+  @AfterClass
+  public void stopProxy() {
+    _mockServer.stop();
+  }
+
+  @AfterMethod
+  public void reset(){
+    _mockServer.reset();
+  }
+
+  @Test
+  public void mustBuildTunnelAndStartAcceptingConnections()
```

Comment:
I'll see if they can. My current concern is that if this spawns threads and we run too many tests in parallel, we may cause Travis to kill the process.